timemachine/cpp/src/kernels/k_nonbonded_pair_list.cuh

```diff
@@ -154,8 +153,16 @@ void __global__ k_nonbonded_pair_list(
         gj_y += FLOAT_TO_FIXED_NONBONDED(-delta_prefactor * delta_y);
         gj_z += FLOAT_TO_FIXED_NONBONDED(-delta_prefactor * delta_z);
 
-        // energy is size extensive so this may not be a good idea
-        energy += FLOAT_TO_FIXED_NONBONDED(u);
+        if (u_buffer) {
+            // Don't add energy if it overflows, but always add the overflow count
+            int overflow_count = 0;
+            if (!energy_overflowed<RealType>(u, overflow_count)) {
+                // energy is size extensive so this may not be a good idea
+                unsigned long long energy = FLOAT_TO_FIXED_NONBONDED(u);
```

Comment:
Could you elaborate on this comment? 

(Do we use the same `FIXED_EXPONENT` for the accumulator (which represents a size-extensive) as for the individual pair interactions (which are not size-extensive)?)

Reply:
The comment is probably stale/copy-pasta. It doesn't apply to the pair interactions specifically, rather to the`u_buffer` which accumulates N energies into a final value. Inclined to remove as this seems confusing to mention in the pair list when it really applies more broadly to how we accumulate energies. 

We do use the same `FIXED_EXPONENT` in the final accumulation for converting back to double. https://github.com/proteneer/timemachine/blob/master/timemachine/cpp/src/fixed_point.hpp#L16-L18 https://github.com/proteneer/timemachine/blob/feature/detect-nb-clashes/timemachine/cpp/src/wrap_kernels.cpp#L74